plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryPageSink.java

```diff
@@ -64,28 +73,46 @@ public BigQueryPageSink(
         this.pageSinkIdColumnName = requireNonNull(pageSinkIdColumnName, "pageSinkIdColumnName is null");
         checkArgument(temporaryTableName.isPresent() == pageSinkIdColumnName.isPresent(),
                 "temporaryTableName.isPresent is not equal to pageSinkIdColumn.isPresent");
-        this.tableId = temporaryTableName
-                .map(tableName -> TableId.of(remoteTableName.getProjectId(), remoteTableName.getDatasetName(), tableName))
-                .orElseGet(remoteTableName::toTableId);
+        this.tableName = temporaryTableName
+                .map(tableName -> TableName.of(remoteTableName.getProjectId(), remoteTableName.getDatasetName(), tableName))
+                .orElseGet(remoteTableName::toTableName);
     }
 
     @Override
     public CompletableFuture<?> appendPage(Page page)
     {
-        InsertAllRequest.Builder batch = InsertAllRequest.newBuilder(tableId);
+        JSONArray batch = new JSONArray();
         for (int position = 0; position < page.getPositionCount(); position++) {
-            Map<String, Object> row = new HashMap<>();
+            JSONObject row = new JSONObject();
             pageSinkIdColumnName.ifPresent(column -> row.put(column, pageSinkId.getId()));
             for (int channel = 0; channel < page.getChannelCount(); channel++) {
                 row.put(columnNames.get(channel), readNativeValue(columnTypes.get(channel), page.getBlock(channel), position));
             }
-            batch.addRow(row);
+            batch.put(row);
         }
 
-        client.insert(batch.build());
+        insertWithCommitted(batch);
         return NOT_BLOCKED;
     }
 
+    private void insertWithCommitted(JSONArray batch)
+    {
+        WriteStream stream = WriteStream.newBuilder().setType(COMMITTED).build();
```

Comment:
we are creating a new stream per Page.
BigQuery has a limit of 1k streams open at a time and 30k/4 hours (7.5k creations per hour).

Let's either document this if we think it's shouldn't be a problem or let's consider creating a single stream per page-sink/tablewriter task.

Also do we consider to use "pending mode" long term to provide proper isolation? With current mode ("committed") if a single stream fails then writes from other streams will still succeed and be visible. i.e. it's not ACID and there's no way to rollback. Note that this is same behaviour that we already had so it's not a regression in that sense.